src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1541,6 +1584,9 @@ public function getParameter(string $name): array|bool|string|int|float|\UnitEnu
         }
 
         if (!(isset($this->parameters[$name]) || isset($this->loadedDynamicParameters[$name]) || \array_key_exists($name, $this->parameters))) {
+            if ('.' === $name[0] && isset($this->getRemovedParameters()[$name])) {
+                throw new \LogicException(sprintf('The parameter "%s" is only accessible at build-time and has been removed.', $name));
```

Comment:
`ParameterNotFoundException` would be better to me

Reply:
Unfortunately this exception can use a custom message as it takes the parameter name as arg.
So it's not clear why the param is not found.
Maybe a `LogicException` from the DI component, which I've missed that it's already used in the dumped container?